modules/users/manifests/telegraf.pp

```diff
@@ -0,0 +1,23 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class users::telegraf {
+    $username = 'telegraf'
+    $home = $::operatingsystem ? {
+        Darwin  => "/Users/${username}",
+        default => "/home/${username}"
+    }
+    $group = $::operatingsystem ? {
+        Darwin  => 'staff',
+        Ubuntu  => undef,
+        default => $username
+    }
+
+    user {
+        'telegraf':
+            # password   => '*',  # none allowed
+            groups  => $group,
```

Comment:
The 'gid' attrib should probably be used instead of 'groups' since it is the primary group for the user.

Reply:
thx. I switched to using gid.